### DISCUSSION TITLE
Travis-CI python3.6 Tox launching virtualenv error

### DISCUSSION THREAD

**Discussion by Wei-1** [tox-dev/tox#2286] [Q&A] [answered]
We are using `tox` for py36 test environment setup through `travis-ci`, the tests start breaking in 2021-10-23.

The error that we are getting is:

```
$ tox
GLOB sdist-make: /home/travis/build/apache/incubator-marvin/python-toolbox/setup.py
py36 create: /home/travis/build/apache/incubator-marvin/python-toolbox/.tox/py36
ERROR: invocation failed (exit code 1), logfile: /home/travis/build/apache/incubator-marvin/python-toolbox/.tox/py36/log/py36-0.log
================================== log start ===================================
StopIteration: 
=================================== log end ====================================
ERROR: InvocationError for command /home/travis/virtualenv/python3.6.7/bin/python -m virtualenv --no-download --python /home/travis/virtualenv/python3.6.7/bin/python py36 (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py36: InvocationError for command /home/travis/virtualenv/python3.6.7/bin/python -m virtualenv --no-download --python /home/travis/virtualenv/python3.6.7/bin/python py36 (exited with code 1)
The command "tox" exited with 1.
```

[Full Log Travis-CI Link](https://app.travis-ci.com/github/apache/incubator-marvin/jobs/549545377)

The log showed that the process had failed while launching `virtualenv` even before package installation.

We had also tested with `tox -rvve py36` for more logs:

[Another log with Travis-CI Link](https://app.travis-ci.com/github/apache/incubator-marvin/jobs/549619263)

The test result showed that `/home/travis/virtualenv/python3.6.7/bin/python` does exist, and can be successfully executed.

What is weird to us is that the same setting used to be able to run without failure in 2021-10-22:

[Previous log with Travis-CI Link](https://app.travis-ci.com/github/apache/incubator-marvin/jobs/544594850)

And the only difference seem to be the version change of `virtualenv`, which is changing from `20.8.1` to `20.9.0` or `20.10.0`.

Does anyone know how this can be fixed?

**Comment by gaborbernat** [comment]
Can you please amend your travis file to show the tox run in verbose mode? tox -vvv that is.

**Comment by Wei-1** [comment]
After adding the verbose, I actually don't see much different in the log though.
https://app.travis-ci.com/github/apache/incubator-marvin/jobs/550859698

**Comment by gaborbernat** [comment]
You could try to invoke the failing command directly and see what gives:

```
/home/travis/virtualenv/python3.6.7/bin/python -m virtualenv --no-download --python /home/travis/virtualenv/python3.6.7/bin/python py36
```

(add -vvv for that verbose mode too). Also please check the virtualenv version. Based on this I can conclude that 99% of this is a broken virtualenv/python distribution on Travis, and is not otherwise related to tox.

**Comment by Wei-1** [comment, marked as answer]
There are several dependencies that could be the cause of the issue.
https://github.com/apache/incubator-marvin/pull/59
In our test, we ended up change the Ubuntu version for the right OpenSSL dependency.
